src/quantem/imaging/time_series.py

```diff
@@ -0,0 +1,255 @@
+        else:
+            pad_width = np.array(pad_width)
+            if pad_width.size == 1:
+                pad_width = np.array(
+                    [pad_width,pad_width],
+                )
+
+        # apply padding
+        padded = self.pad(
+            pad_width = ((0,0),(0,2 * pad_width[0]),(0,2 * pad_width[1])),
+            modify_in_place = modify_in_place
+        )
+
+        # initialize empty array
+        stack_pad = pad_val * np.ones(padded.shape)
+
+        # edge blend values
+        if edge_blend is not None:
+            edge_blend = np.array(edge_blend)
+            
+            if edge_blend.size == 1:
+                edge_blend = np.array(
+                    [edge_blend,edge_blend],
+                )
+            
+            # create tukey window function for blending
+            wx = tukey(self.shape[1],alpha=2*edge_blend[0]/self.shape[1])[None,:,None]
+            wy = tukey(self.shape[2],alpha=2*edge_blend[1]/self.shape[2])[None,None,:]
+            window = wx * wy
+
+            # combine tukey window and images
+            stack_pad[
+                :, 
+                pad_width[0]:stack_pad.shape[1]-pad_width[0], 
+                pad_width[1]:stack_pad.shape[2]-pad_width[1]] = \
+                self.array * window + pad_val * (1 - window)
+        else:
+            # add image to stack
+            stack_pad[
+                :, 
+                pad_width[0]:stack_pad.shape[1]-pad_width[0], 
+                pad_width[1]:stack_pad.shape[2]-pad_width[1]] = \
+                self.array 
+
+        # ensure edge blend values are within range
+        if edge_blend[0] < 0 or edge_blend[0] > stack_pad.shape[1] / 2:
+            if edge_blend[1] < 0 or edge_blend[1] > stack_pad.shape[2] / 2:
+                raise ValueError("edge_blend is outside of allowable range.")
+        
+        return stack_pad
+        
+    def align_images(
+        im_ref: [NDArray, Any],
+        im: [NDArray, Any],
+        return_aligned_image = True,
+        window_size: [int, None] = 7,            
+    ):
+        """
+        Using 2D DFT cross correlation alignment on two images.
+
+        Parameters
+        ----------
+        im_ref: [NDArray, Any]
+            The first image to use as reference
+        im: [NDArray, Any]
+            The second image to shift with respect to the reference
+        return_aligned_image = True
+            If True, returns the shifted second image
+        window_size: [int, None] = 7
+            The size of the window around the subpixel position. If None, defaults to 7
+
+        Returns
+        -------
+        shift: NDArrray
+            shifted coordinates (x,y) between two images
+        im_shift: NDArray
+            shifted image with respect to the shifted coordinates
+        """
+        # take 2D DFT of an image (im) and the image before it as reference (im_ref)
+        G_ref = np.fft.fft2(im_ref) 
+        G = np.fft.fft2(im) 
```

Comment:
It's usually not recommended to use a single letter for variable names in Python.